splice/web/api/campaign.py

```diff
@@ -0,0 +1,156 @@
+
+
+class CampaignListAPI(Resource):
+    def __init__(self):
+        self.reqparse = campaign_parser
+
+        super(CampaignListAPI, self).__init__()
+
+    def get(self):
+        """Returns all the campaigns.
+
+        Takes an optional account_id as a query string argument.
+        """
+        campaigns = get_campaigns(request.args.get('account_id'))
+        return {'results': marshal(campaigns, campaign_fields)}
+
+    def post(self):
+        """Creates a campaign."""
+        args = self.reqparse.parse_args()
+        try:
+            with session_scope() as session:
+                new = insert_campaign(session, args)
+        except IntegrityError:
+            return {'message': 'Invalid account_id or channel_id provided.'}, 400
+        else:
+            return {'result': marshal(new, campaign_fields)}, 201
+
+
+class CampaignAPI(Resource):
+    def __init__(self):
+        self.reqparse = campaign_parser.copy()
+
+        # Mostly the same parser. Name, channel, account and status are
+        # optional here.
+        self.reqparse.replace_argument('name', type=unicode, required=False,
+                                       help='Name of the campaign',
+                                       location='json', store_missing=False)
+        self.reqparse.replace_argument('channel_id', type=int, required=False,
+                                       help='Channel ID', location='json',
+                                       store_missing=False)
+        self.reqparse.replace_argument('account_id', type=int, required=False,
+                                       help='Account ID', location='json',
+                                       store_missing=False)
+        self.reqparse.replace_argument('status', type=str, required=False,
+                                       help='Status', location='json',
+                                       store_missing=False)
+
+        super(CampaignAPI, self).__init__()
+
+    def get(self, campaign_id):
+        """Returns the campaign with given campaign_id."""
+        campaign = get_campaign(campaign_id)
+        if campaign is None:
+            return {'message': 'Campaign with id={id} not found.'.format(
+                id=campaign_id)}, 404
+        else:
+            return {'result': marshal(campaign, campaign_fields)}
+
+    def put(self, campaign_id):
+        """updates the campaign with given campaign_id."""
+        args = self.reqparse.parse_args()
+        try:
+            with session_scope() as session:
+                campaign = update_campaign(session, campaign_id, args)
+        except NoResultFound as e:
+            return {'message': e.message}, 404
+        except IntegrityError as e:
+            return {'message': 'Invalid account_id or channel_id provided.'}, 400
+
+        return {'result': marshal(campaign, campaign_fields)}, 200
+
+
+api.add_resource(CampaignListAPI, '/campaigns', endpoint='campaigns')
```

Comment:
Rather than passing the account_id as an argument here, why not use '/account/<int:account_id>/campaigns' here? Is there any advantage by doing that, or do you have other concerns? For other models, we've decided to build the endpoint with all the IDs in the URL. So we'd better be consistent.